seqlike/SeqLike.py

```diff
@@ -216,7 +214,11 @@ def aa(self, auto_translate=True, **kwargs) -> "SeqLike":
         if self._nt_record and self._aa_record is None and auto_translate:
             translate_kwargs = dict(id=True, name=True, description=True, annotations=True, dbxrefs=True)
             translate_kwargs.update(kwargs)
-            return self.translate(**translate_kwargs)
+            translated = self.translate(**translate_kwargs)
+            # neutralize "protein" `molecule_type` annotation added by BioPython's `SeqRecord.translate()`
+            # translated._aa_record.annotations.pop("molecule_type")
+            translated._aa_record.annotations["molecule_type"] = None
```

Comment:
I considered popping "molecule_type" from `annotations`, but this caused the annotations check in the `test_SeqLike.py` unittest to fail.  Setting "molecule_type" to `None` seems compatible with `SeqRecord`.

Reply:
I think we should pop it, do it in self.translate(), and change the test.  I think the only reason that test fails is because we had to previously change it to account for wonky molecule_type handling.

Reply:
Done